### PR TITLE
Add support for GL_NV_displacement_micromap.

### PR DESCRIPTION
**Pull request by alelenv**
SPV Headers : https://github.com/KhronosGroup/SPIRV-Headers/pull/374
GLSL : https://github.com/KhronosGroup/GLSL/pull/213
SPV : https://github.com/KhronosGroup/SPIRV-Registry/pull/217

### CODE REVIEW

**Comment by arcady-lunarg** [comment]
Does this depend on a change to spirv-tools (for the SPIR-V validator)? If so, that will need to be merged first and known_good.json updated so that the tests can pass. I can review the code changes here in the meantime however.

**Comment by alelenv** [comment]
@arcady-lunarg : Nothing required in validator since just adds a few builtins and decorations. Can updated if anything else is expected
Yes will required headers to be merged before we can flip known_good

Thanks for reviewing

**Comment by alelenv** [comment]
@arcady-lunarg : I have updated known_good and all tests are passing
Please re-review

Thanks

**Comment by alelenv** [comment]
@arcady-lunarg : Gentle one week ping!
Thanks

**Review by ncesario-lunarg — COMMENTED** [review]
This LGTM aside from the spirv-tools update, which will need to happen separately. @jeremy-lunarg I think we are planning on updating this soon?

File: SPIRV/GlslangToSpv.cpp [hunk 1]
```diff
@@ -3650,13 +3672,18 @@ bool TGlslangToSpvTraverser::visitAggregate(glslang::TVisit visit, glslang::TInt
     } else if (node->getOp() == glslang::EOpRayQueryGetIntersectionTriangleVertexPositionsEXT) {
         std::vector<spv::IdImmediate> idImmOps;
 
+        glslang::TOperator rqOp = node->getOp();
```
Comment (arcady-lunarg) [inline SPIRV/GlslangToSpv.cpp:3675]: This variable appears to be unused, could you remove it?

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
Just one minor nit (the unused variable), otherwise this looks good.

File: SPIRV/GlslangToSpv.cpp (hunk 1 above)
Comment (alelenv) [reply SPIRV/GlslangToSpv.cpp]: Done, thanks for the catch!

**Comment by alelenv** [comment]
@arcady-lunarg : Please approve and merge

**Review by arcady-lunarg — APPROVED** [review]
LGTM, thanks.

**Pull request merged**